com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/executors/SqlQueryExecutor.java

```diff
@@ -17,8 +17,9 @@ public SqlQueryExecutor(DataSource dataSource) {
     }
 
     public <R> R executeWithAutoCommit(SqlQuery<R> sqlQuery) throws SQLException {
-        Connection connection = dataSource.getConnection();
+        Connection connection = null;
         try {
+            connection = dataSource.getConnection();
```

Comment:
Why do we need to initialize the connection in the try block?

Reply:
Because it throws SQLException and will not be handled properly?